File: app/src/main/java/com/lofts/dictionary/adapter/DefineAdapter.java
package com.lofts.dictionary.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import com.lofts.dictionary.R;
import com.lofts.dictionary.bean.Define;
import com.lofts.dictionary.bean.Sample;

import java.util.ArrayList;
import java.util.List;

public class DefineAdapter extends BaseAdapter {

    private Context mContext;
    private List<Define> mListDefine;

    public DefineAdapter(Context context) {
        mContext = context;
        mListDefine = new ArrayList<>();
    }

    public void resetList(List<Define> list) {
        mListDefine.clear();
        mListDefine.addAll(list);
        notifyDataSetChanged();
    }

    @Override
    public int getCount() {
        return mListDefine.size();
    }

    @Override
    public Object getItem(int position) {
        return mListDefine.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        convertView = LayoutInflater.from(mContext).inflate(R.layout.item_define, parent, false);

        TextView type = convertView.findViewById(R.id.textview_type);
        TextView paraphrase = convertView.findViewById(R.id.textview_paraphrase);

        Define define = mListDefine.get(position);
        type.setText(define.getPos());
        paraphrase.setText(define.getDef());

        return convertView;
    }
}

File: app/src/main/java/com/lofts/dictionary/adapter/DefineAdapter.kt
package com.lofts.dictionary.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView
import com.lofts.dictionary.R
import com.lofts.dictionary.bean.Define

/**
 * @name      DictionaryKotlin
 * @package   com.lofts.dictionary.adapter
 * @author    Administrator
 * @time      2018/12/19 17:19
 * @describe
 */
class DefineAdapter(context: Context) : BaseAdapter() {

    private var mContext: Context
    private var mListDefine: MutableList<Define>

    init {
        mContext = context
        mListDefine = ArrayList()
    }

    fun resetList(list: List<Define>?){
        mListDefine.clear()
        mListDefine.addAll(list!!)
        notifyDataSetChanged()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
        //var convertView = convertView
        var convertView = LayoutInflater.from(mContext).inflate(R.layout.item_define, parent, false)

        var type = convertView.findViewById<TextView>(R.id.textview_type)
        val paraphrase = convertView.findViewById<TextView>(R.id.textview_paraphrase)

        val define = mListDefine[position]
        type.text = define.pos
        paraphrase.text = define.def

        return convertView
    }

    override fun getItem(position: Int): Any {
        return mListDefine.get(position)
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getCount(): Int {
        return mListDefine.size
    }


}